[Term Entry] PHP Array term array_combine() (#2675)
Codecademy/docs: content/php/concepts/arrays/terms/array-combine/array-combine.md

---
Title: '.array_combine()'

> Description: 'The array_combine() function returns an array by using the elements of one "keys" array as keys and the elements of another "values" array as the corresponding values.'

Reviewer: ```suggestion
Description: 'Returns a new array by combining two arrays, where one array represents keys and the other array represents values.'
```
Let's keep this short and descriptive we can explain it more down below

Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Arrays'
  - 'Methods'
  - 'Functions'
CatalogContent:
  - 'learn-php'
  - 'paths/computer-science'
---

> The `array_combine()` returns an array by using the elements of one "keys" array as keys and the elements of another "values" array as the corresponding values.

Reviewer 2: ```suggestion
The **`array_combine()`** returns an array by using the elements of one "keys" array as keys and the elements of another "values" array as the corresponding values.
```

## Syntax

```pseudo
array_combine(array $keys, array $values): array
```

If the two arrays do not have an equal number of elements, `array_combine()` will throw a `ValueError` as of PHP 8.0.0.

## Example

> 

Reviewer 2: Have a sentence before the example block summarizing what it does.

```php
$keys = array("one", "two", "three", "four");
$values = array(1, 2, 3, 4);

$combinedArray = array_combine($keys, $values);

print_r($combinedArray);
```

The output looks like this:

```shell
Array
(
    [one] => 1
    [two] => 2
    [three] => 3
    [four] => 4
)
```

> Additionally, if there are multiple keys in the `keys` array, the last value that corresponds to the last occurrence of the key will prevail.

Reviewer 2: Swap this line with the header on line 51.

## Example with Duplicate Keys

```php
$keys = array("one", "two", "two", "four");
$values = array(1, 2, 3, 4);

$combinedArray = array_combine($keys, $values);

print_r($combinedArray);
```

The output looks like this:

```shell
Array
(
    [one] => 1
    [two] => 3
    [four] => 4
)
```

In the example above, the last value of the key `two` is `3`, so the value `3` is used in the combined array.

## Codebyte Example

> 

Reviewer 2: Have a sentence introducing the codebyte and summarizing what it's doing.

---
Title: '.array_combine()'
Description: 'Returns a new array by combining two arrays, where one array represents keys and the other array represents values.'
Subjects:
  - 'Computer Science'
  - 'Web Development'
Tags:
  - 'Arrays'
  - 'Methods'
  - 'Functions'
CatalogContent:
  - 'learn-php'
  - 'paths/computer-science'
---

The **`array_combine()`** returns an array by using the elements of one "keys" array as keys and the elements of another "values" array as the corresponding values.

## Syntax

```pseudo
array_combine(array $keys, array $values): array
```

If the two arrays do not have an equal number of elements, `array_combine()` will throw a `ValueError` as of PHP 8.0.0.

## Example

> In this example, we will create an array with the keys `one`, `two`, `three`, and `four` and the values `1`, `2`, `3`, and `4`, respectively.

Reviewer 2: ```suggestion
In this example, an array is created with the keys `one`, `two`, `three`, and `four` and the values `1`, `2`, `3`, and `4`, respectively.
```
We avoid first- and second-person pronouns in Docs.

```php
$keys = array("one", "two", "three", "four");
$values = array(1, 2, 3, 4);

$combinedArray = array_combine($keys, $values);

print_r($combinedArray);
```

The output looks like this:

```shell
Array
(
    [one] => 1
    [two] => 2
    [three] => 3
    [four] => 4
)
```

Additionally, if there are multiple keys in the `keys` array, the last value that corresponds to the last occurrence of the key will prevail.

```php
$keys = array("one", "two", "two", "four");
$values = array(1, 2, 3, 4);

$combinedArray = array_combine($keys, $values);

print_r($combinedArray);
```

The output looks like this:

```shell
Array
(
    [one] => 1
    [two] => 3
    [four] => 4
)
```

In the example above, the last value of the key `two` is `3`, so the value `3` is used in the combined array.

## Codebyte Example

The following codebyte example creates an array of colors and an array of food. It then combines the two arrays into a new array called `$colorHex` with the colors as keys and the food as values.

```codebyte/php
<?php

>   $colors = array("red", "green", "blue");
>   $food = array("apple", "broccoli", "blueberry");
>   $colorHex = array_combine($colors, $hex);
>   print_r($colorHex);

Reviewer 3: ```suggestion
  $colors = array("red", "green", "blue");
  $food = array("apple", "broccoli", "blueberry");
  $colorAndFood = array_combine($colors, $food);
  print_r($colorAndFood);
```
